utils/common/src/main/java/org/apache/brooklyn/util/text/StringFunctions.java

```diff
@@ -67,6 +70,15 @@ public String apply(@Nullable Object[] input) {
             }
         };
     }
+    
+    /** given e.g. "hello %s %s" returns a function which will insert an Iterable of two strings into that pattern */
+     public static Function<Iterable<?>, String> formatterForIterable(final String pattern) {
+        return new Function<Iterable<?>, String>() {
+            public String apply(@Nullable Iterable<?> input) {
+                return String.format(pattern, Iterables.toArray(input, Object.class));
```

Comment:
This will NPE if `input` is null, yet the parameter is marked `@Nullable`. Should we remove the annotation, or explicitly check for null (e.g. to return null if input is null)?